### ISSUE TITLE
Client randomly crashes when loading media

### ISSUE THREAD

**Issue by raku-cat** [telegramdesktop/tdesktop#6124]
### Steps to reproduce
1. Load media
2. Client freezes and crashes

### Expected behaviour
The client should not crash

### Actual behaviour
The client crashes about half way through loading media

### Configuration
**Operating system:** Arch linux

**Version of Telegram Desktop:** 1.7.6

**Used theme**: Default night theme

<details><summary><b>Logs</b>:</summary>

[2019.06.10 20:04:05] Launched version: 1007006, install beta: [TRUE], alpha: 0, debug mode: [FALSE], test dc: [FALSE]
[2019.06.10 20:04:05] Executable dir: /usr/bin/, name: telegram-desktop
[2019.06.10 20:04:05] Initial working dir: /home/josh/
[2019.06.10 20:04:05] Working dir: /home/josh/.local/share/TelegramDesktop/
[2019.06.10 20:04:05] Command line: telegram-desktop
[2019.06.10 20:04:05] Executable path before check: /usr/bin/telegram-desktop
[2019.06.10 20:04:05] Logs started
[2019.06.10 20:04:05] Fontconfig version: 2.13.1.
[2019.06.10 20:04:05] Custom FONTCONFIG_FILE: /home/josh/.local/share/TelegramDesktop/tdata/fc-custom-1.conf
[2019.06.10 20:04:05] Connecting local socket to /tmp/83d7ff8162350c3503e347e44ea7f1f9-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...
[2019.06.10 20:04:05] This is the only instance of Telegram, starting server and app...
[2019.06.10 20:04:05] Moved logging from '/home/josh/.local/share/TelegramDesktop/log_start0.txt' to '/home/josh/.local/share/TelegramDesktop/log.txt'!
[2019.06.10 20:04:05] Primary screen DPI: 96
[2019.06.10 20:04:09] Could not load 'ayatana-appindicator3' with version 1 :(
[2019.06.10 20:04:09] Error: failed to load 'gdk_x11_drawable_get_xdisplay' function!
[2019.06.10 20:04:09] Getting DE, XDG_CURRENT_DESKTOP: 'i3'
[2019.06.10 20:04:09] Getting DE, DESKTOP_SESSION: 'i3'
[2019.06.10 20:04:09] Getting DE, KDE_SESSION_VERSION: ''
[2019.06.10 20:04:09] Getting DE, GNOME_DESKTOP_SESSION_ID: ''
[2019.06.10 20:04:09] Getting DE, KDE_FULL_SESSION: ''
[2019.06.10 20:04:09] DE: Other
[2019.06.10 20:04:09] Tray Icon: Try Qt = [TRUE], Prefer appindicator = [FALSE]
[2019.06.10 20:04:09] App Info: reading settings...
[2019.06.10 20:04:09] App Info: reading encrypted settings...
[2019.06.10 20:04:09] Lang Info: Loaded cached, keys: 2325
[2019.06.10 20:04:09] OpenAL Logging Level: (not set)
[2019.06.10 20:04:09] Audio Playback Devices: OpenAL Soft
[2019.06.10 20:04:09] Audio Playback Default Device: OpenAL Soft
[2019.06.10 20:04:09] Audio Capture Devices: Yeti Stereo Microphone Analog Stereo;Monitor of GP104 High Definition Audio Controller Digital Stereo (HDMI 2);Monitor of Yeti Stereo Microphone Analog Stereo;Monitor of Built-in Audio Analog Stereo;Built-in Audio Analog Stereo
[2019.06.10 20:04:09] Audio Capture Default Device: Yeti Stereo Microphone Analog Stereo
[2019.06.10 20:04:09] App Info: reading map...
[2019.06.10 20:04:09] App Info: reading encrypted map...
[2019.06.10 20:04:09] App Info: reading encrypted user settings...
[2019.06.10 20:04:09] App Info: encrypted user settings read.
[2019.06.10 20:04:09] App Info: reading encrypted mtp data...
[2019.06.10 20:04:09] MTP Info: read keys, current: 5, to destroy: 0
[2019.06.10 20:04:09] Map read time: 4
[2019.06.10 20:04:09] Notifications Server: Xfce Notify Daemon
[2019.06.10 20:04:09] LibNotify capabilities: actions, body, body-hyperlinks, body-markup, icon-static, x-canonical-private-icon-only
[2019.06.10 20:04:09] Export Info: Destroy top bar by controller removal.
[2019.06.10 20:04:09] Using Qt tray icon.
[2019.06.10 20:04:09] Tray Icon: Using Qt tray icon, available: [TRUE]
[2019.06.10 20:04:09] Not using Unity Launcher count.
[2019.06.10 20:04:45] Skipping message, because it is already in blocks!
[2019.06.10 20:05:12] Skipping message, because it is already in blocks!
[2019.06.10 20:05:13] App Info: writing encrypted user settings...
[2019.06.10 20:06:05] Skipping message, because it is already in blocks!
[2019.06.10 20:06:18] Skipping message, because it is already in blocks!
[2019.06.10 20:06:29] Skipping message, because it is already in blocks!
[2019.06.10 20:06:44] Skipping message, because it is already in blocks!
[2019.06.10 20:09:15] Message Info: bad message notification received (error_code 16) for msg_id = 6701072615721526376, seq_no = 270
[2019.06.10 20:10:08] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:10:10] Audio Info: Closing audio playback device.
[2019.06.10 20:10:41] Skipping message, because it is already in blocks!
[2019.06.10 20:10:59] Skipping message, because it is already in blocks!
[2019.06.10 20:11:06] App Error: Bad photo data for saving to cache.
[2019.06.10 20:11:08] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:11:10] Audio Info: Closing audio playback device.
[2019.06.10 20:11:54] Skipping message, because it is already in blocks!
[2019.06.10 20:12:14] App Error: Bad photo data for saving to cache.
[2019.06.10 20:14:17] Message Info: bad message notification received (error_code 16) for msg_id = 6701073932267618788, seq_no = 484
[2019.06.10 20:14:17] Message Info: bad message notification received (error_code 16) for msg_id = 6701073932267618792, seq_no = 485
[2019.06.10 20:19:21] Message Info: bad message notification received (error_code 16) for msg_id = 6701075228457820448, seq_no = 538
[2019.06.10 20:19:38] Message Info: bad message notification received (error_code 16) for msg_id = 6701075228457820460, seq_no = 542
[2019.06.10 20:24:44] Message Info: bad message notification received (error_code 16) for msg_id = 6701076605710980068, seq_no = 592
[2019.06.10 20:25:02] Skipping message, because it is already in blocks!
[2019.06.10 20:25:10] Skipping message, because it is already in blocks!
[2019.06.10 20:25:23] Skipping message, because it is already in blocks!
[2019.06.10 20:25:29] Skipping message, because it is already in blocks!
[2019.06.10 20:25:36] Skipping message, because it is already in blocks!
[2019.06.10 20:29:47] Message Info: bad message notification received (error_code 16) for msg_id = 6701077920679295964, seq_no = 690
[2019.06.10 20:30:30] Skipping message, because it is already in blocks!
[2019.06.10 20:31:21] Skipping message, because it is already in blocks!
[2019.06.10 20:34:58] Message Info: bad message notification received (error_code 16) for msg_id = 6701079221305352428, seq_no = 820
[2019.06.10 20:35:02] Message Info: bad message notification received (error_code 16) for msg_id = 6701079221305352432, seq_no = 820
[2019.06.10 20:35:05] Message Info: bad message notification received (error_code 16) for msg_id = 6701079221305352444, seq_no = 824
[2019.06.10 20:35:51] Skipping message, because it is already in blocks!
[2019.06.10 20:36:06] App Error: Bad photo data for saving to cache.
[2019.06.10 20:36:06] App Error: Bad photo data for saving to cache.
[2019.06.10 20:36:39] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:36:41] Audio Info: Closing audio playback device.
[2019.06.10 20:36:42] Skipping message, because it is already in blocks!
[2019.06.10 20:37:27] Skipping message, because it is already in blocks!
[2019.06.10 20:38:20] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:38:22] Audio Info: Closing audio playback device.
[2019.06.10 20:40:16] Message Info: bad message notification received (error_code 16) for msg_id = 6701080589106498160, seq_no = 990
[2019.06.10 20:45:19] Message Info: bad message notification received (error_code 16) for msg_id = 6701081924399274904, seq_no = 7
[2019.06.10 20:45:19] Skipping message, because it is already in blocks!
[2019.06.10 20:45:57] App Error: Bad photo data for saving to cache.
[2019.06.10 20:46:12] App Error: Bad photo data for saving to cache.
[2019.06.10 20:46:12] App Error: Bad photo data for saving to cache.
[2019.06.10 20:46:12] App Error: Bad photo data for saving to cache.
[2019.06.10 20:46:13] Skipping message, because it is already in blocks!
[2019.06.10 20:46:39] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:46:41] Audio Info: Closing audio playback device.
[2019.06.10 20:47:29] Skipping message, because it is already in blocks!
[2019.06.10 20:47:29] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:47:31] Audio Info: Closing audio playback device.
[2019.06.10 20:47:33] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:47:35] Audio Info: Closing audio playback device.
[2019.06.10 20:47:35] Skipping message, because it is already in blocks!
[2019.06.10 20:48:15] Skipping message, because it is already in blocks!
[2019.06.10 20:48:18] Skipping message, because it is already in blocks!
[2019.06.10 20:48:54] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:48:56] Audio Info: Closing audio playback device.
[2019.06.10 20:50:25] Message Info: bad message notification received (error_code 16) for msg_id = 6701083226164391248, seq_no = 1260
[2019.06.10 20:50:28] Message Info: bad message notification received (error_code 16) for msg_id = 6701083226164391252, seq_no = 1260
[2019.06.10 20:50:31] Message Info: bad message notification received (error_code 16) for msg_id = 6701083226164391256, seq_no = 1261
[2019.06.10 20:50:31] Message Info: bad message notification received (error_code 16) for msg_id = 6701083226164391260, seq_no = 1263
[2019.06.10 20:50:31] Message Info: bad message notification received (error_code 16) for msg_id = 6701083226164391264, seq_no = 1265
[2019.06.10 20:50:33] Skipping message, because it is already in blocks!
[2019.06.10 20:51:14] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:51:16] Audio Info: Closing audio playback device.
[2019.06.10 20:51:29] Skipping message, because it is already in blocks!
[2019.06.10 20:51:30] Skipping message, because it is already in blocks!
[2019.06.10 20:52:35] Skipping message, because it is already in blocks!
[2019.06.10 20:53:14] Skipping message, because it is already in blocks!
[2019.06.10 20:53:15] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:53:17] Audio Info: Closing audio playback device.
[2019.06.10 20:53:19] Skipping message, because it is already in blocks!
[2019.06.10 20:53:56] Skipping message, because it is already in blocks!
[2019.06.10 20:53:56] Skipping message, because it is already in blocks!
[2019.06.10 20:54:05] Skipping message, because it is already in blocks!
[2019.06.10 20:54:06] Skipping message, because it is already in blocks!
[2019.06.10 20:54:48] Skipping message, because it is already in blocks!
[2019.06.10 20:54:59] Skipping message, because it is already in blocks!
[2019.06.10 20:55:24] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 20:55:25] Audio Info: Closing audio playback device.
[2019.06.10 20:55:34] Message Info: bad message notification received (error_code 16) for msg_id = 6701084567323550520, seq_no = 1500
[2019.06.10 20:55:35] Message Info: bad message notification received (error_code 16) for msg_id = 6701084567323550536, seq_no = 1504
[2019.06.10 20:55:44] Skipping message, because it is already in blocks!
[2019.06.10 20:55:48] Skipping message, because it is already in blocks!
[2019.06.10 20:56:09] Skipping message, because it is already in blocks!
[2019.06.10 20:56:27] Skipping message, because it is already in blocks!
[2019.06.10 20:56:32] Skipping message, because it is already in blocks!
[2019.06.10 20:57:02] Skipping message, because it is already in blocks!
[2019.06.10 20:57:51] Skipping message, because it is already in blocks!
[2019.06.10 20:58:02] Skipping message, because it is already in blocks!
[2019.06.10 20:58:26] Skipping message, because it is already in blocks!
[2019.06.10 20:58:38] Skipping message, because it is already in blocks!
[2019.06.10 20:58:54] Skipping message, because it is already in blocks!
[2019.06.10 20:59:02] Skipping message, because it is already in blocks!
[2019.06.10 20:59:26] Skipping message, because it is already in blocks!
[2019.06.10 20:59:33] Skipping message, because it is already in blocks!
[2019.06.10 20:59:37] Skipping message, because it is already in blocks!
[2019.06.10 21:00:21] Skipping message, because it is already in blocks!
[2019.06.10 21:00:33] Skipping message, because it is already in blocks!
[2019.06.10 21:00:44] Message Info: bad message notification received (error_code 16) for msg_id = 6701085872528776232, seq_no = 1748
[2019.06.10 21:00:46] Message Info: bad message notification received (error_code 16) for msg_id = 6701085872528776236, seq_no = 1748
[2019.06.10 21:00:52] Message Info: bad message notification received (error_code 16) for msg_id = 6701085872528776248, seq_no = 1750
[2019.06.10 21:01:03] Skipping message, because it is already in blocks!
[2019.06.10 21:01:11] Skipping message, because it is already in blocks!
[2019.06.10 21:01:25] Skipping message, because it is already in blocks!
[2019.06.10 21:03:03] Audio Info: recreating audio device and reattaching the tracks
[2019.06.10 21:03:05] Audio Info: Closing audio playback device.
[2019.06.10 21:05:59] Message Info: bad message notification received (error_code 16) for msg_id = 6701087230275748120, seq_no = 1891
[2019.06.10 21:11:11] Message Info: bad message notification received (error_code 16) for msg_id = 6701088552603814948, seq_no = 1934
[2019.06.10 21:11:12] Message Info: bad message notification received (error_code 16) for msg_id = 6701088552603814952, seq_no = 1935
[2019.06.10 21:11:12] Message Info: bad message notification received (error_code 16) for msg_id = 6701088552603814956, seq_no = 1937
[2019.06.10 21:11:47] Skipping message, because it is already in blocks!
[2019.06.10 21:11:53] Skipping message, because it is already in blocks!
[2019.06.10 21:16:17] Message Info: bad message notification received (error_code 16) for msg_id = 6701089896058626180, seq_no = 2082
[2019.06.10 21:16:17] Message Info: bad message notification received (error_code 16) for msg_id = 6701089896058626184, seq_no = 2082
[2019.06.10 21:16:24] Message Info: bad message notification received (error_code 16) for msg_id = 6701089896058626200, seq_no = 2086
[2019.06.10 21:20:10] Skipping message, because it is already in blocks!
[2019.06.10 21:21:35] Message Info: bad message notification received (error_code 16) for msg_id = 6701091234948203296, seq_no = 2148
[2019.06.10 21:21:36] Message Info: bad message notification received (error_code 16) for msg_id = 6701091234948203300, seq_no = 2148
[2019.06.10 21:21:46] Message Info: bad message notification received (error_code 16) for msg_id = 6701091234948203304, seq_no = 2148
[2019.06.10 21:21:47] Message Info: bad message notification received (error_code 16) for msg_id = 6701091234948203308, seq_no = 2148
[2019.06.10 21:21:55] Message Info: bad message notification received (error_code 16) for msg_id = 6701091234948203324, seq_no = 2152
[2019.06.10 21:26:57] Message Info: bad message notification received (error_code 16) for msg_id = 6701092656766090940, seq_no = 2202
[2019.06.10 21:32:08] Message Info: bad message notification received (error_code 16) for msg_id = 6701093952882522012, seq_no = 2254
[2019.06.10 21:32:08] Message Info: bad message notification received (error_code 16) for msg_id = 6701093952882522016, seq_no = 2254
[2019.06.10 21:32:26] Message Info: bad message notification received (error_code 16) for msg_id = 6701093952882522028, seq_no = 2256
[2019.06.10 21:32:26] Message Info: bad message notification received (error_code 16) for msg_id = 6701093952882522032, seq_no = 2257
</details>
This happens completely randomly and has seemed to have happened since 1.7 came out. I have autoloading of everything off, when I click to load GIFs usually, the client will crash, when I open it back up the media is usually loaded, and can play fine. There is no consistency in what will make it do this(no specific image or size), nor does it happen on any regular basis, the client can run fine for a week and then crash.

**Comment by raku-cat** [comment]
As of 1.7.14 this still happens, if I load multiple gifs in a chat I can be basically guaranteed that the client will crash.

**Comment by liolok** [comment]
Same issue here and repeating so frequently: Every single time program tries to show a GIF, it's a risk of crashing.

I've already turned off "**Enable animations**" and "**Autoplay GIFs**" in `Settings/Advanced/Performance/` but nothing changed. It seems like that things went wrong even before playing, but when auto-loading. This also makes sense to the log below.

Environment: `community/telegram-desktop 1.7.14-3` on Arch Linux

<details><summary>Complete Log:</summary><p>

```
[2019.08.03 08:11:48] Launched version: 1007014, install beta: [FALSE], alpha: 0, debug mode: [FALSE], test dc: [FALSE]
[2019.08.03 08:11:48] Executable dir: /usr/bin/, name: telegram-desktop
[2019.08.03 08:11:48] Initial working dir: /home/liolok/
[2019.08.03 08:11:48] Working dir: /home/liolok/.local/share/TelegramDesktop/
[2019.08.03 08:11:48] Command line: telegram-desktop --
[2019.08.03 08:11:48] Executable path before check: /usr/bin/telegram-desktop
[2019.08.03 08:11:48] Logs started
[2019.08.03 08:11:48] Connecting local socket to /tmp/701cd2eb1a6d9830946077540ccdff0e-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...
[2019.08.03 08:11:48] This is the only instance of Telegram, starting server and app...
[2019.08.03 08:11:48] Moved logging from '/home/liolok/.local/share/TelegramDesktop/log_start0.txt' to '/home/liolok/.local/share/TelegramDesktop/log.txt'!
[2019.08.03 08:11:48] Primary screen DPI: 96
[2019.08.03 08:11:48] Font Error: could not resolve 'Open Sans' font, got 'Noto Sans'.
[2019.08.03 08:11:48] Font Error: could not resolve 'Open Sans' font, got 'Noto Sans'.
[2019.08.03 08:11:48] Font Error: could not resolve 'Open Sans Semibold' font, got 'Noto Sans'.
[2019.08.03 08:11:48] Could not load 'ayatana-appindicator3' with version 1 :(
[2019.08.03 08:11:48] Error: failed to load 'gdk_x11_drawable_get_xdisplay' function!
[2019.08.03 08:11:48] Getting DE, XDG_CURRENT_DESKTOP: 'GNOME'
[2019.08.03 08:11:48] Getting DE, DESKTOP_SESSION: 'gnome-xorg'
[2019.08.03 08:11:48] Getting DE, KDE_SESSION_VERSION: ''
[2019.08.03 08:11:48] DE: Gnome
[2019.08.03 08:11:48] Tray Icon: Try Qt = [TRUE], Prefer appindicator = [TRUE]
[2019.08.03 08:11:48] App Info: reading settings...
[2019.08.03 08:11:48] App Info: reading encrypted settings...
[2019.08.03 08:11:48] Lang Info: Loaded cached, keys: 2398
[2019.08.03 08:11:48] Lang Error: Unexpected tag (key 'lng_call_box_status_group:count')
[2019.08.03 08:11:48] Lang Error: Unknown tag (key 'lng_manage_discussion_group_warning:visible')
[2019.08.03 08:11:49] OpenAL Logging Level: (not set)
[2019.08.03 08:11:49] Audio Playback Devices: OpenAL Soft
[2019.08.03 08:11:49] Audio Playback Default Device: OpenAL Soft
[2019.08.03 08:11:49] Audio Capture Devices: USB PnP Audio Device Analog Stereo;Monitor of USB PnP Audio Device Analog Stereo;Monitor of Built-in Audio Analog Stereo;Built-in Audio Analog Stereo
[2019.08.03 08:11:49] Audio Capture Default Device: USB PnP Audio Device Analog Stereo
[2019.08.03 08:11:49] App Info: reading map...
[2019.08.03 08:11:49] App Info: reading encrypted map...
[2019.08.03 08:11:49] App Info: reading encrypted user settings...
[2019.08.03 08:11:49] App Info: encrypted user settings read.
[2019.08.03 08:11:49] App Info: reading encrypted mtp data...
[2019.08.03 08:11:49] MTP Info: read keys, current: 5, to destroy: 0
[2019.08.03 08:11:49] Map read time: 8
[2019.08.03 08:11:49] Notifications Server: gnome-shell
[2019.08.03 08:11:49] LibNotify capabilities: actions, body, body-markup, icon-static, persistence, sound
[2019.08.03 08:11:49] Export Info: Destroy top bar by controller removal.
[2019.08.03 08:11:49] Tray Icon: Using appindicator tray icon.
[2019.08.03 08:11:49] Not using Unity Launcher count.
[2019.08.03 08:11:50] Config Error: Could not find Answer in dns response JSON.
[2019.08.03 08:11:51] Not Secure Error: bad request answer start (-1983040163 405586054 -1817042061)
[2019.08.03 08:11:51] Exception: bad pq reply
[2019.08.03 08:11:51] Config Error: Could not find Answer in dns response JSON.
[2019.08.03 08:11:55] Config Error: Could not find Answer in dns response JSON.
[2019.08.03 08:11:56] Config Error: Could not find Answer in dns response JSON.
[2019.08.03 08:14:52] App Error: Bad photo data for saving to cache.
[2019.08.03 08:14:52] App Error: Bad photo data for saving to cache.
[2019.08.03 08:14:52] App Error: Bad photo data for saving to cache.
[2019.08.03 08:14:52] App Error: Bad photo data for saving to cache.
[2019.08.03 08:14:52] App Error: Bad photo data for saving to cache.
```

</p></details>


Apparently the key line is `App Error: Bad photo data for saving to cache.`

**Comment by eiglow** [comment]
i get this issue too, on telegram-desktop 1.8.4, on both KDE Neon and Fedora

**Comment by Aokromes** [comment]
plz, the next time you get a crash press send crash report and post crash id here.

**Comment by Aokromes** [comment]
no answer after >2 months, i suppose this issue was fixed, if no i suggest you to test on 1.9.6 since it fixes some leaks.
